### PR TITLE
Ph update telemetry locig

### PR DESCRIPTION
**Pull request by phendriksen100**
- Cleaned up Sample class *ReadGroup on another PR
- Added validation for the xml files and will throw an error when not formatted proper
- Cleaned up logic for calling telemetry rerport

### CODE REVIEW

File: src/resources/test_data_read_group.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
+[
```
Comment (sahakiann) [inline src/resources/test_data_read_group.json:1]: Should this and the other new json file be committed to the repo? Or were they just for testing?

File: src/resources/test_data_sample.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[
+    {
+        "attributes": {
+            "aggregation_project": "RP-1545",
+            "location": "OnPrem",
+            "md5": "gs://fc-d40787e1-06e7-40e7-af19-134b48dd88a1/RP-1545/Exome/16177_CCPM_1300040_BLOOD/v4/16177_CCPM_1300040_BLOOD.bam.md5",
+            "hard_clipping_used": False,
```
Comment (sahakiann) [inline src/resources/test_data_sample.json:7]: If these are going to stay in the repo - this should be `false` to be valid JSON

File: src/scripts/dbgap/dbgap_classes.py [hunk 3]
```diff
@@ -49,55 +49,47 @@ def __init__(self, json_object, md5):
...
-        self.study = root[0].attrib
-        self.bio_project = bio_projects[0]
-        self.dbgap_sample_info = samples[0]
+        return os.path.splitext(aggregation_path)[1][1:]
 
+    @property
+    def dbgap_info(self):
+        if self._dbgap_info is None:
+            self._dbgap_info = DbgapTelemetryWrapper(phs_id=self.phs).get_sample_info(alias=self.alias)
+        return self._dbgap_info
```
Comment (sahakiann) [inline src/scripts/dbgap/dbgap_classes.py:78]: So it looks like you're calling `self._set_sample_attributes` in the init method, which is setting `self._dbgap_info` to None. But then in the `dbgap_info` property, you check if it's None, and set it to the `DbgapTelemetryWrapper` instance. I think you can probably skip all of that, and set `self.dbgap_info` in the init method directly. 

So something like: 

```
def __init__():
    self._set_sample_attributes(sample_json, sample_id, md5)
    self.dbgap_info = DbgapTelemetryWrapper(phs_id=self.phs).get_sample_info(alias=self.alias)
```

File: src/scripts/dbgap/dbgap_classes.py [hunk 4]
```diff
@@ -240,20 +232,20 @@ def __init__(self, sample, read_group):
...
+        formatted_data_type = self.sample.formatted_data_type["constant"].replace(" ", "_")
 
         return f"{self.sample.phs}.{pdo_or_wr}.{self.read_group.library_name}.{pairing_code}.{self.sample.alias}.{self.sample.project}.{formatted_data_type}.{self.sample.version}"
 
     def get_file_name(self):
         return f"{self.get_submitter_id()}.add.experiment.xml"
 
     def get_title(self):
-        repo = self.sample.get_biospecimen_repo()
+        repo = self.sample.biospecimen_repo
```
Comment (sahakiann) [inline src/scripts/dbgap/dbgap_classes.py:243]: Looks like `repo` could potentially be an empty string. Does the return statement need to get adjusted in the event that it is an empty string?

File: src/services/dbgap_telemetry_report.py [hunk 5]
```diff
@@ -23,28 +23,38 @@ def call_telemetry_report(self):
 
         return xmltodict.parse(response.text)
 
-    def get_sample_status(self, alias, data_type):
+    def _get_sample(alias):
```
Comment (sahakiann) [inline src/services/dbgap_telemetry_report.py:26]: Looks like `self` needs to be added back, and `call_telemetry_report` should be `self._ call_telemetry_report()`

```
    def _get_sample(self, alias):
        try:
            telemetry_data = self._call_telemetry_report()
```

File: src/scripts/dbgap/dbgap_classes.py (hunk 3 above)
Comment (phendriksen100) [reply src/scripts/dbgap/dbgap_classes.py:78]: Yea, I was trying to be clever and use lazy loading lol. Maybe it is just more confusing, I will switch it up

**Review by sahakiann — APPROVED** [review]

**Pull request merged**